node/src/components/chain_synchronizer/operations.rs

```diff
@@ -725,13 +725,25 @@ pub(super) async fn run_chain_sync_task(
 
     let maybe_last_emergency_restart_era_id = chainspec.protocol_config.last_emergency_restart;
     if let Some(last_emergency_restart_era) = maybe_last_emergency_restart_era_id {
-        if last_emergency_restart_era > trusted_block_header.era_id() {
+        if last_emergency_restart_era > trusted_block_header.next_block_era_id() {
```

Comment:
Maybe it's just me but `trusted_block_header.next_block_era_id() < ` would be easier 😅 

Also, can you add a comment that we have to increment the era id b/c there will be a no-block era with a switch block created just after the emeregency upgrade?

Reply:
Done in https://github.com/casper-network/casper-node/pull/2580/commits/544c44454fe753821eb4343d5a0d395286a02345.